Log levels adjusted.

kbayraktar opened the pull request.

kbayraktar committed:
Log levels adjusted.

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -110,20 +110,24 @@ public HostDriverFixture(Connection connection) {
     @FixtureMethod
     public boolean connect(String s3270Path, String hostname, int port, int offsetRow, int offsetColumn) throws FixtureException {
-        logger.info("Host-Fixture connecting ...");
+        logger.debug("Host-Fixture connecting ...");
```
I'd add the parameters of the fixture method to the debug output

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -110,20 +110,24 @@ public HostDriverFixture(Connection connection) {
             return true;
         } else {
-            logger.info("The connection to host '" + hostname + "' on port '" + port + "' could not be established.");
+            logger.error("The connection to host '" + hostname + "' on port '" + port + "' could not be established.");
```
warn

gunther-bachmann replied:
throw FixtureException instead of returning false

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -168,7 +172,7 @@ public boolean disconnect() throws FixtureException {
     @FixtureMethod
     public Status getStatus() throws FixtureException {
-        logger.info("get Status ...");
+        logger.trace("get Status ...");
```
debug

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -322,6 +332,7 @@ private Result setCursorPosition(int row, int col) {
      */
     @FixtureMethod
     public Result moveCursor(String elementLocator, LocatorStrategy locatorType) throws FixtureException {
+        logger.trace("Move cursor to position {} by {}.",elementLocator, locatorType );
```
debug level

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -368,6 +379,7 @@ public Result moveCursor(String elementLocator, LocatorStrategy locatorType) thr
      */
     @FixtureMethod
     public String buildAllFieldsAsString() throws FixtureException {
+        logger.trace("Generate all 3270-Fields on screen as String ...");
```
debug level

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -110,20 +110,24 @@ public HostDriverFixture(Connection connection) {
+                + "\n Connected Hostname : {} ;\n Connected Port : {}" , type , charSet, s3270Path, hostname, port);
+        
         timer.startTimer();
         connection.connect(s3270Path, hostname, port, type, terminalMode, charSet, offset);
```
try catch and throw FixtureException

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -406,14 +419,14 @@ private void takeScreenshot(String filenameBase) {
             FileUtils.mkdir(new File(filePath), true);
         } catch (IOException e) {
-            logger.error("Something went wrong while creating test files: ", e);
+            logger.error("Something went wrong while creating screenshot for file {} :", filePath + filename );
```
warning would be better, right. the test may succeed ...

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -442,22 +455,22 @@ public void reported(SemanticUnit unit, Action action, String message, String id
-	}
+    @Override
+    public void reportFixtureExit(FixtureException fixtureException) {
+    // TODO Auto-generated method stub
```
remove todo, add something like
// do nothing
We could however think about writing screenshots here, too.

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -442,22 +455,22 @@ public void reported(SemanticUnit unit, Action action, String message, String id
-	}
+    @Override
+    public void reportExceptionExit(Exception exception) {
+        // TODO Auto-generated method stub
```
see above

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -442,22 +455,22 @@ public void reported(SemanticUnit unit, Action action, String message, String id
+    @Override
+    public void reportAssertionExit(AssertionError assertionError) {
+        // TODO Auto-generated method stub
+        
```
see above

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/locators/LocatorByStart.java:
```diff
@@ -132,21 +140,29 @@ private void initializeStartRowAndColumn(String elementLocator) {
-            throw new IllegalArgumentException(columnExceptionMessage);
+            String columnExceptionMessage = "Your chosen column '" + startColumn + 
+                    "' is greater than the maximum column '" + (maxColumn - 1) + "'";
+            logger.error(columnExceptionMessage);
```
error logging is not necessary if throwing a fixture exception

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/locators/LocatorByStartStop.java:
```diff
@@ -59,14 +64,12 @@
+     * @throws FixtureException 
      */
-    public LocatorByStartStop(int startRow, int startColumn, int endRow, int endColumn, Status status, Offset offset) {
+    public LocatorByStartStop(int startRow, int startColumn, int endRow, int endColumn, Status status, Offset offset) throws FixtureException {
```
no need for FixtureException

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -133,9 +133,9 @@ public boolean disconnect() {
+            logger.trace("Disconnected successfully from host : {} ", hostname);
         } else {
-            logger.info("Discoonection failed");
+            logger.error("Disconnection failed");
```
throw fixture exception instead of logging errors

kbayraktar commented:
Please Can you add these lines to build.gradle file:
```javascript
// For a summary of failed tests at the end of test execution.
tasks.withType(Test) {
    // a collection to track failedTests
    ext.failedTests = []
     afterTest { descriptor, result ->
        if (result.resultType == TestResult.ResultType.FAILURE) {
            String failedTest = "${descriptor.className}::${descriptor.name}"
            logger.debug("Adding " + failedTest + " to failedTests...")
            failedTests << [failedTest]
        }
    }
     afterSuite { suite, result ->
        if (!suite.parent) { // will match the outermost suite
            // logs each failed test
            if (!failedTests.empty) {
                logger.lifecycle("\n\n*************************************************************************************************")
                logger.lifecycle("\nFailed tests:")
                failedTests.each { failedTest ->
                    logger.lifecycle("${failedTest}")
                }
            }
            logger.lifecycle("\n*************************************************************************************************\n")
        }
    }
}
```

kbayraktar committed:
Pull Request changes added.

kbayraktar committed:
Changes made for proof of certification.
Because of failing integration tests, the connection to the host with
certificates were necessary.

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -107,28 +107,50 @@ public HostDriverFixture(Connection connection) {
+     * @param verifyCertificate 
+     *            This boolean value can be true or false. 
+     *            <p>If <b><i>true</i></b> is chosen, several environment variables on test system can be set as follows:
+     *            <br>CADIR : Specifies a directory containing CA (root) certificates to use when verifying a certificate provided by the host. (OpenSSL only). 
```
prefix with 'HOSTFIXTURE_'

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -254,21 +276,27 @@ protected String getElement(String elementLocator, LocatorStrategy locatorType)
+            }
+            
+        } catch (Exception e) {
+            throw new FixtureException(e.getMessage(), FixtureException.keyValues("elementlocator", elementLocator, 
```
pass e as parameter

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/HostDriverFixture.java:
```diff
@@ -329,26 +357,33 @@ private Result setCursorPosition(int row, int col) {
+            }
+            
+        } catch (Exception e) {
+            throw new FixtureException(e.getMessage(), FixtureException.keyValues("elementlocator", elementLocator, 
```
pass e as parameter to FixtureException

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/locators/LocatorByStartStop.java:
```diff
@@ -181,7 +179,7 @@ public int getEndRowWithOffset() {
      *            in the form "1;2;4;5"
      * @throws FixtureException 
```
@throws is no longer valid

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -111,31 +121,71 @@ public Connection connect(String s3270Path, String hostname, int port, TerminalT
+            } else if (KEYPASSWD != null && KEYPASSWD.length() > 0) {
+                buffer.append(" -keypasswd" + KEYPASSWD);
+            } 
+        }else {
```
return buffer.toString();

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -111,31 +121,71 @@ public Connection connect(String s3270Path, String hostname, int port, TerminalT
+                buffer.append(" -keypasswd" + KEYPASSWD);
+            } 
+        }else {
+            commandLine = commandLine + " -noverifycert";
```
return commandLine + " -noverifycert";

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -111,31 +121,71 @@ public Connection connect(String s3270Path, String hostname, int port, TerminalT
+        }else {
+            commandLine = commandLine + " -noverifycert";
+        }
+        return commandLine;
```
remove this line

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -111,31 +121,71 @@ public Connection connect(String s3270Path, String hostname, int port, TerminalT
             s3270Process.waitFor(5, TimeUnit.SECONDS);
         } catch (final InterruptedException ex) {
-            logger.error("Something went wrong during termination of s2370 process");
+            logger.error("Something went wrong during termination of s3270 process");
```
should be warning

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/net/Connection.java:
```diff
@@ -111,31 +121,71 @@ public Connection connect(String s3270Path, String hostname, int port, TerminalT
             in.close();
         } catch (final IOException ex) {
-            logger.error("Something went wrong during closing InputStreamreader of s2370 process");
+            logger.error("Something went wrong during closing InputStreamreader of s3270 process");
```
warning

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/util/LineReader.java:
```diff
@@ -106,12 +105,12 @@ public String readSingleLineWidth(String line, LocatorByWidth locator) throws Fi
                 result = lineReplaced.substring(startColumn, startColumn + locator.getWidth());
             } catch (StringIndexOutOfBoundsException e) {
-                throw new FixtureException(
+                throw new RuntimeException(
```
IllegalArgument?

gunther-bachmann commented on src/main/java/org/testeditor/fixture/host/util/LineReader.java:
```diff
@@ -106,12 +105,12 @@ public String readSingleLineWidth(String line, LocatorByWidth locator) throws Fi
             }
         } else {
-            throw new FixtureException("Your chosen argument width '" + locator.getWidth()
+            throw new RuntimeException("Your chosen argument width '" + locator.getWidth()
```
IllegalArgument?

kbayraktar committed:
Pull-Request changes added.

kbayraktar committed:
Obsolete import deleted.
